syntax = "proto3";
package qa;

service QAServer {
  rpc Answer(Question) returns (AnswerResponse) {}
}

message Question {
  string question = 1;
  string paper_id = 2; // optional
}

message AnswerResponse {
  string answer = 1;
  double confidence = 2;
  double retrieval_ms = 3;
  double inference_ms = 4;
  double end_to_end_ms = 5;
}
